mantidproject/mantid
# Add reduction call metadata to ISIS ORSO file

### Description of work

#### Summary of work

Adds the reduction call information (the third bullet point on the issue) to the ISIS ORSO file metadata. This is the series of high-level Mantid algorithm calls that were performed for the reduction, as stored in the workspace history.



Refs #36395. 




#### Further detail of work

Some of the string data is being output in the ORSO file with single quotes. I have looked into this and it seems to be something that is done by the PyYAML library that orsopy is using to write the file. Apparently this library will put single quotes around values that it would otherwise find ambiguous when trying to read the file back in and determine the type. For the reduction call data specifically, the single quotes seem to be added when there are blank lines in the script. I've tested and we are passing the data in as a string and when it's parsed with the single quotes removed it's read back in as a string, so it's unclear why the quotation marks are being added in this particular case. As it's an issue with a library rather than in Mantid itself we cannot address it in our code.

On the tests, I have extended the existing tests to cover the new functionality, however I am working on maintenance issue #36777 to improve the overall approach to these.

### To test:



1) Go to `Interfaces -> Reflectometry -> ISIS Reflectometry`.
1) On the Runs tab, enter Investigation ID `1120015` and Cycle `11_3`. Click the Search button.
1) From the search results, click to highlight runs `11934`, `11935` and `11936` and click the Transfer button to transfer them into the batch table on the right hand side. Click the Process button to reduce the data.
1) Go to the Save ASCII tab. Set a Save Path for saving the file to.
1) In the File Format section, select `ORSO ASCII (*.ort)` from the drop-down.
1) From the List of Workspaces section on the left, select workspace `IvsQ_11934_11935+11936` to highlight it.
1) Click the big Save button on the right hand side to save to file. There should be a successful call to `SaveISISReflectometryORSO` recorded in the messages pane. If you navigate to the save path that you set in the Save ASCII tab, you should see a `.ort` file. You should be able to open this in a text editor and see that under the `Reduction -> call` heading in the file header there is an entry showing the parent algorithm calls from the reduced workspace history.





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
